### ISSUE TITLE
COM Server always runs with `latin-1` codec

### ISSUE THREAD

**Issue by elonzh** [mhammond/pywin32#1688]
![屏幕截图 2021-04-08 210339](https://user-images.githubusercontent.com/8749411/114032504-f1990180-98ae-11eb-82e5-7b17dce37724.png)

For all bugs, please provide the following information.

* Expected behavior and actual behavior.
	
	make non-ASCII output without crash

* Steps to reproduce the problem.

	write a COM add-in and trigger make some non-ASCII output

* Version of Python and pywin32

	- CPython v3.8.9
	- pywin32 300(amd64)

**Comment by mhammond** [comment]
That looks like whereever the "print" is going is failing (eg, the vscode terminal window?) At line 222, try printing `type(msg)` - I expect it will show it is unicode.

**Comment by elonzh** [comment]
> That looks like whereever the "print" is going is failing (eg, the vscode terminal window?) At line 222, try printing `type(msg)` - I expect it will show it is unicode.

`type(msg)` is `str` and It will always crash wherever the display window we use(eg, PythonWin window)

**Comment by mhammond** [comment]
That's surprising in Python 3 - OnConnection() doesn't have an argument named `msg` - where does it come from? Can you paste your entire `OnConnection` method?

**Comment by kxrob** [comment]
> As you can see, `sys.getdefaultencoding` and `sys.getfilesystemencoding` are `utf-8` but the `print` function is still use `latin-1` codec.


The `print() / logger ... traceback.print_exception(... sys.stderr)`  uses the `.encoding` (and `.errors` tolerance) of the sys.stdout / .stderr (if present), independent of the `sys.getdefaultencoding()` etc.

Normal interactive or VSCode terminal:

```
>>> sys.stdout
<_io.TextIOWrapper name='<stdout>' mode='w' encoding='utf-8'>
>>> sys.stdout.encoding
'utf-8'
>>> sys.stderr
<_io.TextIOWrapper name='<stderr>' mode='w' encoding='utf-8'>
>>> sys.stderr.errors
'backslashreplace'
```

PythonWin:

```
>>> sys.stdout
<pywin.framework.interact.DockedInteractiveView object at 0x0000019CFFBBA1C8>
>>> sys.stdout.encoding
'utf-8'
>>> 
```

=> You may inspect what `sys.stdout / .stderr (.encoding)` you have in your sitation and why.
=> In certain cases, if not a problem of win32com, it may help to install a custom (tolerant) IO wrapper into `sys.stdout / .stderr`

**Comment by elonzh** [comment]
@kxrob I already tried this but it won't work, because `sys.stdout` and `sys.stderr` are `PyTraceObject` which does not have an `encoding` attribute.

**Comment by kxrob** [comment]
> does not have an `encoding` attribute

That may be the problem and need a fix. It can be set / modified directly there also so far. 

-->
https://github.com/mhammond/pywin32/blob/e0a7d99d7fe770e6eda58c1fa0988e7cc48f9b97/win32/src/win32trace.cpp#L111-L120
...


(The makeshift bypass wrapper may need to be errors tolerant for latin-1 meanwhile - but not sure...)

**Comment by mhammond** [comment]
I suspect this was fixed by d3b91862b9ce9147fffba4aea9debb3a4df7cba1 - please reopen if you can still reproduce a problem once 304 is released.
